Advent of `tests/ui` (misc cleanups and improvements) [1/N]

jieyouxu opened the pull request:
Part of #133895.

Misc improvements to some ui tests immediately under `tests/ui/`.

Best reviewed commit-by-commit.

Thanks @clubby789 for PR title suggestion 😸.

r? compiler

jieyouxu committed:
Adjust `alias-uninit-value.rs`

- Document and tidy up `alias-uninit-value.rs`
- Move `alias-uninit-value.rs` to `tests/ui/codegen/`

jieyouxu committed:
Adjust `allow-non-lint-warnings.rs`

- Document `allow-non-lint-warnings.rs`
- Move `allow-non-lint-warnings.rs` under `tests/ui/diagnostic-flags/`
- Improve the test to use two *differential* revisions:
    1. One revision checks that without `-A warnings` the code sample
       actually emits a warning.
    2. The other revision checks that `-A warnings` suppresses the
       warning.
  This makes sure that if the code sample no longer warns, the test
  doesn't silently pass but fail to check its intended purpose.

jieyouxu committed:
Adjust `anonymous-higher-ranked-lifetime.rs`

- Document `anonymous-higher-ranked-lifetime.rs`
- Move `anonymous-higher-ranked-lifetime.rs` to `tests/ui/higher-ranked`

jieyouxu committed:
Adjust `artificial-block.rs`

- Document `artificial-block.rs`
- Move `artificial-block.rs` under `tests/ui/reachable`

jieyouxu commented on tests/ui/as-precedence.rs:
This is downgraded to check-pass because this is a parser precedence check.

jieyouxu replied:
Actually I guess I should not downgrade this because the assert might come in handy

jieyouxu replied:
Reverted this change.

jieyouxu commented on tests/ui/diagnostic-flags/allow-non-lint-warnings.rs:
```diff
@@ -0,0 +1,20 @@
+// ignore-tidy-linelength
+//! Check that `-A warnings` cli flag applies to *all* warnings, including feature gate warnings.
+//!
+//! This test tries to exercise that by checking that the "empty trait list for derive" warning for
+//! `#[derive()]` is permitted by `-A warnings`, which is a non-lint warning.
```
Remark: I elaborated on the "how" here because it was very not obvious how this test did the check.

jieyouxu commented on tests/ui/codegen/alias-uninit-value.rs:
Discussion: this one I have no idea. I suspect this test in the current form is not checking what the original commit intended to check.

jieyouxu replied:
Original form as in fb76c25729ae6cf2c755d7515830a27f9b0971ec is:

```rs
tag sty {
    ty_nil;
}

type raw_t = rec(sty struct,
                 option::t[str] cname,
                 uint hash);

fn mk_raw_ty(sty st, &option::t[str] cname) -> raw_t {
  ret rec(struct=st,
	  cname=cname,
	  hash=0u);
}

fn main() {
  mk_raw_ty(ty_nil, none[str]);
}
```

jieyouxu committed:
Adjust `as-precedence.rs`

- Document `as-precedence.rs`
- Move `as-precedence.rs` under `tests/ui/parser/`

fmease commented on tests/ui/as-precedence.rs:
Lol, interesting corner case of the language. I never thought about operator precedence in code like `&() as &Trait + Send` (Rust 2015): It gets parsed as `&() as &(Trait) + Send` instead of `&() as &(Trait + Send)` and therefore `Send` doesn't get interpreted as a trait object trait bound.

fmease reviewed (approved):
Thanks for your patience. r=me with concerns resolved and renaming-commits squashed into the respective previous commit.

jieyouxu commented on tests/ui/alias-uninit-value.rs:
I probably would, I didn't change the casing for this particular test because I had *no* idea what it was trying to check lol.

jieyouxu replied:
Fixed non camel case types to remove distraction.

jieyouxu commented on tests/ui/allow-non-lint-warnings.rs:
```diff
@@ -1,3 +1,15 @@
+// ignore-tidy-linelength
+//! Check that `-A warnings` cli flag applies to *all* warnings, including feature gate warnings.
```
Good point, I'll fix up this mention.

jieyouxu replied:
Removed.

jieyouxu commented on tests/ui/allow-non-lint-warnings.rs:
```diff
@@ -1,3 +1,15 @@
+// ignore-tidy-linelength
+//! Check that `-A warnings` cli flag applies to *all* warnings, including feature gate warnings.
+//!
+//! This test tries to exercise that by checking that the "empty trait list for derive" warning for
```
Oh wait, very good catch, I missed that this was changed.

jieyouxu replied:
Indeed, this was changed as part of https://github.com/rust-lang/rust/pull/79078

jieyouxu replied:
1. Changed the test to use a code sample that actually emits a non-lint warning.
2. Make the test use **differential revisions**: one revision checks that without `-A warnings` the code sample does actually emit a warning, and another revision checks that the warning is suppressed if `-A warnings` cli flag is specified. This makes sure that the code sample under test doesn't silently pass yet fail to check that non-lint warnings can be suppressed by `-A warnings` if the code sample no longer emits any non-lint warnings to be suppressed.

jieyouxu commented on tests/ui/artificial-block.rs:
```diff
@@ -1,5 +1,30 @@
+//! Check that we don't get compile errors on unreachable code after the `{ ret 3; }` artificial
```
This was a copy-pasta 😆 I'll fix it.

jieyouxu replied:
Fixed

jieyouxu commented on tests/ui/artificial-block.rs:
```diff
@@ -1,5 +1,30 @@
+//! reduce to build-pass or even check-pass.
+//!
+//! This test was introduced as part of commit `a833f152baa17460e8414355e832d30d5161f8e8` which
+//! removes an "artificial block". See also commit `8d381823e2aa1524eabeb3219d7dc1d5007e6096` for
```
You are 100% correct. I somehow conjured a *completely* unrelated **full** commit hash into this comment, the correct hash is actually 3d738e9e0634a4cd6239d1317bd7dad53be68dc8, thankfully I reproduced the commit message otherwise I would not have been able to find it.

jieyouxu replied:
Fixed

jieyouxu commented:
Rebase force-push, no functional changes

jieyouxu commented:
Squash force-push, no functional changes, squashed move commits into the test content changes for each test.

jieyouxu commented:
### Changes since last review

Diff: https://github.com/rust-lang/rust/compare/32b9336c54b0f2b42ca37f1132979cee3c0030b5..56efa0a7bf55aa4d1413008df672b95f09a4d8e9

1. `tests/ui/codegen/alias-uninit-value.rs`: fixed non camel case types to remove distraction.
2. `tests/ui/diagnostic-flags/allow-non-lint-warnings.rs`:
    1. Changed the test to use a code sample that actually emits a non-lint warning.
    2. Make the test use **differential revisions**: one revision checks that without `-A warnings` the code sample does actually emit a warning, and another revision checks that the warning is suppressed if `-A warnings` cli flag is specified. This makes sure that the code sample under test doesn't silently pass yet fail to check that non-lint warnings can be suppressed by `-A warnings` if the code sample no longer emits any non-lint warnings to be suppressed.
3. `tests/ui/reachable/artificial-block.rs`:
    1. Fix comment code snippet to use modern syntax.
    2. Fix the incorrect commit hash 8d381823e2aa1524eabeb3219d7dc1d5007e6096 when I actually meant 3d738e9e0634a4cd6239d1317bd7dad53be68dc8.
    3. Various typo/rewording.

Commits are squashed and rebased before force-push 56efa0a7bf55aa4d1413008df672b95f09a4d8e9 so help with the compare diff, they contain no changes beforehand.

fmease reviewed (approved):
Beautiful, thanks a lot!

> differential revisions

:pray: :pray: :pray:

fmease commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 56efa0a7bf55aa4d1413008df672b95f09a4d8e9 has been approved by `fmease`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=fmease 56efa0a7bf55aa4d1413008df672b95f09a4d8e9 -->
<!-- homu: {"type":"Approved","sha":"56efa0a7bf55aa4d1413008df672b95f09a4d8e9","approver":"fmease","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
